docs/agent/hostname_force_config_as_canonical.md

```diff
@@ -0,0 +1,8 @@
+Agent hostname config as canonical 
+
+If `hostname` is set in `datadog.yaml` and the value starts with `ip-` or `domu`, this hostname is not used as a canonical hostname.
+More information about what a canonical hostname is can be found at [How does Datadog determine the Agent hostname?](https://docs.datadoghq.com/agent/faq/how-datadog-agent-determines-the-hostname/?tab=agentv6v7#agent-versions).
+
+To use an hostname starting with `ip-` or `domu` as a canonical hostname, set the config `hostname_force_config_as_canonical` at `true` in datadog.yaml.
```

Comment:
```suggestion
Starting with Agent v6.16.0 and v7.16.0, the Agent supports the config option `hostname_force_config_as_canonical` (default: `false`). When set to `true`, a configuration-provided hostname starting with `ip-` or `domu` will be accepted as the canonical hostname in-app.
```

Reply:
Also, I'd mention here that starting with v6.16.0 and v7.16.0, the Agent logs a warning if it detects a situation where the config-provided hostname is a valid hostname but will not be accepted as the canonical hostname in-app. (and mention the exact warning that's logged)